src/Avalonia.Controls/Platform/Win32Properties.cs

```diff
@@ -14,8 +14,14 @@ namespace Avalonia.Controls
     /// <summary>
     /// Set of Win32 specific properties and events that allow deeper customization of the application per platform.
     /// </summary>
-    public static class Win32Properties
+    public class Win32Properties
     {
+        public static readonly AttachedProperty<Win32WindowCornerHints> WindowCornerHintProperty =
+            AvaloniaProperty.RegisterAttached<Win32Properties, Window, Win32WindowCornerHints>("WindowCornerHint");
+
```

Comment:
Why can't we just make this a 3-state bool value? This property is ONLY controlling a window corner radius. This means we need to control whether it is ON or OFF. Then it's probably useful to have a "default" condition as well.

bool? could communicate:
 - null : Platform default
 - true : Round corners
 - false : Don't round corners

Will there ever be a reason to have more values?

Reply:
If it's a windows specific property, as it is now, it should mirror windows API.
If we assume it shouldn't be windows specific one day, it can either be made a 3-state bool later. But it's unlikely to be supported on wider range of platforms.

Reply:
> If it's a windows specific property, as it is now, it should mirror windows API.

Good point. That's the right way to look at it I think so your probably right.

I also notice there are more values supported: https://learn.microsoft.com/en-us/windows/win32/api/dwmapi/ne-dwmapi-dwm_window_corner_preference. Namely `RoundSmall`. So an enum is the way to go.